### ISSUE TITLE
CAN issues with the lilygo and Volvo XC40 - Polestar 2	78kWh

### ISSUE THREAD

**Issue by GoSmartEV** [dalathegreat/Battery-Emulator#677]
When 12V power is applied to the battery and the LilyGo board, there is occasionally an issue where the CAN bus and the transceiver in the BMS control unit do not "wake up," making communication impossible. So far, I’ve found only one solution to this problem — adding an external MCP2515 module and configuring it for communication with the BMS. In this setup, everything works smoothly and without any issues, ensuring reliable data transmission over the CAN bus.

![photo_5404697660746755858_y](https://github.com/user-attachments/assets/0ad215c0-8165-4978-befc-d14b171c1cf8)

**Comment by lenvm** [comment]
For our understanding, can you please clarify what setup does not work and what setup does work?

Some questions:
- You mention "When 12V power is applied to the battery and the Lilygo", is this the case in both the working setup and the non working setup?
- Have you tried a setup where you are powering the Lilygo with power via USB?

**Comment by GoSmartEV** [comment]
> For our understanding, can you please clarify what setup does not work and what setup does work?
> 
> Some questions:
> 
> * You mention "When 12V power is applied to the battery and the LilyGo", is this the case in both the working setup and the non working setup?
> * Have you tried a setup where you are providing the LilyGo with power via USB?

Yes and yes, there is a problem exactly with the lilygo this is 100% info

**Comment by dalathegreat** [comment]
Closing this issue as completed, should be fixed now with PR https://github.com/dalathegreat/Battery-Emulator/pull/744
